Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -623,6 +645,36 @@ private extension HTTPClient {
 // MARK: - Request Retry Logic
 extension HTTPClient {
 
+    /// Evaluates whether a request should be retried with the next path in the list of fallback paths.
+    ///
+    /// This function checks the HTTP response status code to determine if the request should be retried
+    /// with the next fallback path. If the retry conditions are met, it schedules the request immediately and
+    /// returns `true` to indicate that the request was retried.
+    ///
+    /// - Parameters:
+    ///   - request: The original `HTTPClient.Request` that may need to be retried.
+    ///   - httpURLResponse: An optional `HTTPURLResponse` that contains the status code of the response.
+    /// - Returns: A Boolean value indicating whether the request was retried.
+    internal func retryRequestWithNextFallbackPathIfNeeded(
```

Comment:
Again, nothing to do in this PR, but I was thinking it would be great to have a way to test this more easily. One thing that came to mind would be to add an option to our test proxy to block only these endpoints in the main host...

Reply:
Oh I like that! In fact I'm trying to add these fallback endpoints to our backend tests and I'm really struggling so far 😅